pyinfra/operations/opkg.py

```diff
@@ -0,0 +1,96 @@
+"""
+    Manage packages on OpenWrt using opkg
+        + ``update`` - update local copy of package information
+        + ``packages`` -  install and remove packages
+
+    see https://openwrt.org/docs/guide-user/additional-software/opkg
+    OpenWrt recommends against upgrading all packages  thus there is no ``opkg.upgrade`` function
+"""
+from typing import List, Union
+
+from pyinfra import host
+from pyinfra.api import StringCommand, operation
+from pyinfra.facts.opkg import Packages
+from pyinfra.operations.util.packaging import ensure_packages
+
+EQUALS = "="
+UPDATE_ENTRY = "$$updated$$"  # use $ as not allowed in package names (TODO - need reference)
+
+
+@operation(is_idempotent=False)
+def update(force: bool = False):
+    """
+    Update the local opkg information.  Unless force is set, will only run
+    once per host per invocation of pyinfra.
+
+    + force - refresh the package list even if already done
+    """
+
+    if force or (UPDATE_ENTRY not in host.get_fact(Packages)):
+        host.get_fact(Packages).update({UPDATE_ENTRY: [UPDATE_ENTRY]})
```

Comment:
Note: this won’t work in v3 since facts are no longer cached (fixes so many other issues). You could use host data to achieve the same though (compatible in both 2/3).